test/e2e/network/network_policy.go

```diff
@@ -108,7 +108,7 @@ var _ = SIGDescribe("NetworkPolicy [LinuxOnly]", func() {
 
 			// Create a pod with name 'client-cannot-connect', which will attempt to communicate with the server,
 			// but should not be able to now that isolation is on.
-			testCannotConnect(f, f.Namespace, "client-cannot-connect", service, 80)
+			testCannotConnect(f, f.Namespace, "client-cannot-connect", service, 80, 3)
```

Comment:
Why do some of the tests have retries and some don't?

There's no reason you can't just always retry; the retries would only be used if either (a) they were needed, or (b) the test was failing. They don't slow down the "normal" case where the connection attempt gets the right result on the first try.

Reply:
The test without retries are the ones that do not have a network policy created. I was thinking that retries would not be needed for those cases, but I can change it to just retry, for up to 2 times (3 total attempts before marking as failure) for all of the testCanConnect() and testCannontConnect() if you think that would be better.

Reply:
Revisiting this, I went ahead and made changes so that all of the test will retry up to 2 times on failed attempts (up to 3 total attempts for failing tests), even those without network policy.